### DISCUSSION TITLE
QHY Camera not detected

### DISCUSSION THREAD

**Discussion by Astrokashyap** [aaronwmorris/indi-allsky#1116] [General]
I selected QHY during the setup but I can't detect my QHY 5III 462  on the web application, all the indi-allsky service are connected to the indiserver. Am i missing something???
![qhy](https://github.com/aaronwmorris/indi-allsky/assets/157868841/3fd6c366-e9fd-406f-af4c-2e1e1af6e11d)

**Comment by aaronwmorris** [comment]
If you could run the `./misc/support_info.sh` script and post the output here, it would be helpful.  My guess is the firmware for the QHY camera is not being loaded, which I have a solution.

**Reply by Astrokashyap** [reply]
thanks for helping me, here is the output :-


sky@Titis:~/indi-allsky $ ./misc/support_info.sh
./misc/support_info.sh: line 49: warning: command substitution: ignored null byte in input
#################################
sky@Titis:~/indi-allsky $
#################################

Distribution: Debian
Release: 11
Arch: aarch64
Bits: 64

CPUs: 4
Memory: 7999812 kB

System: Raspberry Pi 4 Model B Rev 1.5

Uptime
 08:24:05 up 43 min,  1 user,  load average: 0.25, 0.20, 0.36

Filesystems
Filesystem     1K-blocks    Used Available Use% Mounted on
/dev/root      122620052 7549588 110055344   7% /
devtmpfs         3735680       0   3735680   0% /dev
tmpfs            3999904     172   3999732   1% /dev/shm
tmpfs            1599964    1112   1598852   1% /run
tmpfs               5120       4      5116   1% /run/lock
/dev/mmcblk0p1    261108   31380    229728  13% /boot
tmpfs             799980       0    799980   0% /run/user/1000

system python: Python 3.9.2

indiserver: /usr/local/bin/indiserver


User info
uid=1000(sky) gid=1000(sky) groups=1000(sky),4(adm),20(dialout),24(cdrom),27(sudo),29(audio),44(video),46(plugdev),60(games),100(users),104(input),106(render),108(netdev),997(gpio),998(i2c),999(spi)

Process info
sky         4083  0.1  0.2  27312 19520 ?        SNs  08:21   0:00 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/python3 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/gunicorn --config /etc/indi-allsky/gunicorn.conf.py indi_allsky.wsgi
sky         4085  7.6  2.6 950128 214844 ?       SNl  08:21   0:13 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/python3 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/gunicorn --config /etc/indi-allsky/gunicorn.conf.py indi_allsky.wsgi
sky         4344  0.0  0.0   9100  2872 ?        Ss   08:22   0:00 /usr/local/bin/indiserver -p 7624 indi_simulator_telescope indi_qhy_ccd
sky         4345  4.5  1.5 500852 122096 ?       SNsl 08:22   0:03 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/python3 allsky.py --log syslog run
sky         4346  0.0  0.1  34660 10124 ?        S    08:22   0:00 indi_simulator_telescope
sky         4347  0.0  0.1 125904 12240 ?        Sl   08:22   0:00 indi_qhy_ccd
sky         4356  0.0  1.0 356872 81104 ?        S    08:22   0:00 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/python3 allsky.py --log syslog run
sky         4357  0.0  1.0 368756 82892 ?        SN   08:22   0:00 /home/sky/indi-allsky/virtualenv/indi-allsky/bin/python3 allsky.py --log syslog run

USB info
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 1618:0462 Cypress WestBridge
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

USB Permissions
      123      0 drwxr-xr-x   4 root     root           80 Jan  1  1970 /dev/bus/usb
      126      0 drwxr-xr-x   2 root     root           60 Jan  1  1970 /dev/bus/usb/002
      127      0 crw-rw-r--   1 root     root     189, 128 Jan 28 07:40 /dev/bus/usb/002/001
      124      0 drwxr-xr-x   2 root     root          100 Jan 28 08:20 /dev/bus/usb/001
      442      0 crw-rw-rw-   1 root     root     189,   2 Jan 28 08:20 /dev/bus/usb/001/003
      138      0 crw-rw-r--   1 root     root     189,   1 Jan 28 07:40 /dev/bus/usb/001/002
      125      0 crw-rw-r--   1 root     root     189,   0 Jan 28 07:40 /dev/bus/usb/001/001

Module info
Module                  Size  Used by
cmac                   16384  3
algif_hash             16384  1
aes_arm64              16384  3
aes_generic            36864  1 aes_arm64
algif_skcipher         16384  1
af_alg                 28672  6 algif_hash,algif_skcipher
bnep                   24576  2
hci_uart               53248  1
btbcm                  24576  1 hci_uart
bluetooth             565248  26 hci_uart,btbcm,bnep
ecdh_generic           16384  2 bluetooth
ecc                    36864  1 ecdh_generic
libaes                 16384  3 aes_arm64,bluetooth,aes_generic
8021q                  32768  0
garp                   16384  1 8021q
stp                    16384  1 garp
llc                    16384  2 stp,garp
vc4                   331776  4
brcmfmac              331776  0
snd_soc_hdmi_codec     20480  2
brcmutil               24576  1 brcmfmac
drm_display_helper     16384  1 vc4
cec                    53248  1 vc4
drm_dma_helper         20480  1 vc4
cfg80211              925696  1 brcmfmac
drm_kms_helper        204800  3 drm_dma_helper,vc4
bcm2835_isp            28672  0
bcm2835_codec          49152  0
bcm2835_v4l2           45056  0
snd_soc_core          274432  2 vc4,snd_soc_hdmi_codec
rpivid_hevc            53248  0
bcm2835_mmal_vchiq     36864  3 bcm2835_codec,bcm2835_v4l2,bcm2835_isp
v3d                    90112  0
v4l2_mem2mem           45056  2 bcm2835_codec,rpivid_hevc
videobuf2_vmalloc      16384  1 bcm2835_v4l2
raspberrypi_hwmon      16384  0
videobuf2_dma_contig    20480  3 bcm2835_codec,rpivid_hevc,bcm2835_isp
gpu_sched              49152  1 v3d
videobuf2_memops       16384  2 videobuf2_vmalloc,videobuf2_dma_contig
rfkill                 32768  6 bluetooth,cfg80211
drm_shmem_helper       24576  1 v3d
videobuf2_v4l2         32768  5 bcm2835_codec,bcm2835_v4l2,rpivid_hevc,v4l2_mem2mem,bcm2835_isp
snd_compress           20480  1 snd_soc_core
videobuf2_common       69632  9 bcm2835_codec,videobuf2_vmalloc,videobuf2_dma_contig,videobuf2_v4l2,bcm2835_v4l2,rpivid_hevc,v4l2_mem2mem,videobuf2_memops,bcm2835_isp
snd_pcm_dmaengine      20480  1 snd_soc_core
i2c_brcmstb            16384  0
videodev              274432  7 bcm2835_codec,videobuf2_v4l2,bcm2835_v4l2,videobuf2_common,rpivid_hevc,v4l2_mem2mem,bcm2835_isp
snd_bcm2835            28672  0
snd_pcm               139264  5 snd_bcm2835,snd_soc_hdmi_codec,snd_compress,snd_soc_core,snd_pcm_dmaengine
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
vc_sm_cma              36864  2 bcm2835_mmal_vchiq,bcm2835_isp
snd_timer              36864  1 snd_pcm
snd                   106496  6 snd_bcm2835,snd_soc_hdmi_codec,snd_timer,snd_compress,snd_soc_core,snd_pcm
mc                     61440  7 videodev,bcm2835_codec,videobuf2_v4l2,videobuf2_common,rpivid_hevc,v4l2_mem2mem,bcm2835_isp
fb_sys_fops            16384  1 drm_kms_helper
uio_pdrv_genirq        16384  0
nvmem_rmem             16384  0
uio                    24576  1 uio_pdrv_genirq
drm                   581632  8 gpu_sched,drm_kms_helper,drm_dma_helper,v3d,vc4,drm_shmem_helper,drm_display_helper
fuse                  135168  1
drm_panel_orientation_quirks    28672  1 drm
backlight              24576  1 drm
ip_tables              32768  0
x_tables               53248  1 ip_tables
ipv6                  557056  29

git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

git log
commit e0c96fc7aedf796751381fbd12655c1fb7edce04
Merge: 1f1c0796 ebb5aecd
Author: Aaron W Morris <aaron@aarmor.net>
Date:   Fri Jan 26 15:53:58 2024 -0600

    Merge pull request #1115 from aaronwmorris/dev

    Misc updates

indi version: 2.0.3

indi packages

Detected indi properties
looking for *.*.*
Connected to localhost on port 7624
Queried properties from *
Telescope Simulator.CONNECTION.CONNECT=Off
Telescope Simulator.CONNECTION.DISCONNECT=On
Telescope Simulator.DRIVER_INFO.DRIVER_NAME=Telescope Simulator
Telescope Simulator.DRIVER_INFO.DRIVER_EXEC=indi_simulator_telescope
Telescope Simulator.DRIVER_INFO.DRIVER_VERSION=1.0
Telescope Simulator.DRIVER_INFO.DRIVER_INTERFACE=5
Telescope Simulator.POLLING_PERIOD.PERIOD_MS=250
Telescope Simulator.DEBUG.ENABLE=Off
Telescope Simulator.DEBUG.DISABLE=On
Telescope Simulator.CONFIG_PROCESS.CONFIG_LOAD=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_SAVE=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_DEFAULT=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_PURGE=Off
Telescope Simulator.CONNECTION_MODE.CONNECTION_SERIAL=On
Telescope Simulator.CONNECTION_MODE.CONNECTION_TCP=Off
Telescope Simulator.DEVICE_PORT.PORT=/dev/ttyUSB0
Telescope Simulator.DEVICE_BAUD_RATE.9600=On
Telescope Simulator.DEVICE_BAUD_RATE.19200=Off
Telescope Simulator.DEVICE_BAUD_RATE.38400=Off
Telescope Simulator.DEVICE_BAUD_RATE.57600=Off
Telescope Simulator.DEVICE_BAUD_RATE.115200=Off
Telescope Simulator.DEVICE_BAUD_RATE.230400=Off
Telescope Simulator.DEVICE_AUTO_SEARCH.INDI_ENABLED=On
Telescope Simulator.DEVICE_AUTO_SEARCH.INDI_DISABLED=Off
Telescope Simulator.DEVICE_PORT_SCAN.Scan Ports=Off
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_GPS=GPS Simulator
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_DOME=Dome Simulator
Telescope Simulator.DOME_POLICY.DOME_IGNORED=On
Telescope Simulator.DOME_POLICY.DOME_LOCKS=Off
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_FOCAL_LENGTH=900
Telescope Simulator.TELESCOPE_INFO.GUIDER_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.GUIDER_FOCAL_LENGTH=900
Telescope Simulator.SCOPE_CONFIG_NAME.SCOPE_CONFIG_NAME=
Telescope Simulator.MOUNT_TYPE is write-only
Telescope Simulator.SIM_PIER_SIDE is write-only
Telescope Simulator.MOUNT_MODEL is write-only
Telescope Simulator.MOUNT_AXES.PRIMARY=0
Telescope Simulator.MOUNT_AXES.SECONDARY=0
Telescope Simulator.FLIP_HA is write-only
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_GPS=GPS Simulator
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_DOME=Dome Simulator
Telescope Simulator.DOME_POLICY.DOME_IGNORED=On
Telescope Simulator.DOME_POLICY.DOME_LOCKS=Off
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_FOCAL_LENGTH=900
Telescope Simulator.TELESCOPE_INFO.GUIDER_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.GUIDER_FOCAL_LENGTH=900
Telescope Simulator.SCOPE_CONFIG_NAME.SCOPE_CONFIG_NAME=
Telescope Simulator.MOUNT_TYPE is write-only
Telescope Simulator.SIM_PIER_SIDE is write-only
Telescope Simulator.MOUNT_MODEL is write-only
Telescope Simulator.MOUNT_AXES.PRIMARY=0
Telescope Simulator.MOUNT_AXES.SECONDARY=0
Telescope Simulator.FLIP_HA is write-only
Telescope Simulator.CONNECTION.CONNECT=Off
Telescope Simulator.CONNECTION.DISCONNECT=On
Telescope Simulator.DRIVER_INFO.DRIVER_NAME=Telescope Simulator
Telescope Simulator.DRIVER_INFO.DRIVER_EXEC=indi_simulator_telescope
Telescope Simulator.DRIVER_INFO.DRIVER_VERSION=1.0
Telescope Simulator.DRIVER_INFO.DRIVER_INTERFACE=5
Telescope Simulator.POLLING_PERIOD.PERIOD_MS=250
Telescope Simulator.DEBUG.ENABLE=Off
Telescope Simulator.DEBUG.DISABLE=On
Telescope Simulator.CONFIG_PROCESS.CONFIG_LOAD=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_SAVE=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_DEFAULT=Off
Telescope Simulator.CONFIG_PROCESS.CONFIG_PURGE=Off
Telescope Simulator.CONNECTION_MODE.CONNECTION_SERIAL=On
Telescope Simulator.CONNECTION_MODE.CONNECTION_TCP=Off
Telescope Simulator.DEVICE_PORT.PORT=/dev/ttyUSB0
Telescope Simulator.DEVICE_BAUD_RATE.9600=On
Telescope Simulator.DEVICE_BAUD_RATE.19200=Off
Telescope Simulator.DEVICE_BAUD_RATE.38400=Off
Telescope Simulator.DEVICE_BAUD_RATE.57600=Off
Telescope Simulator.DEVICE_BAUD_RATE.115200=Off
Telescope Simulator.DEVICE_BAUD_RATE.230400=Off
Telescope Simulator.DEVICE_AUTO_SEARCH.INDI_ENABLED=On
Telescope Simulator.DEVICE_AUTO_SEARCH.INDI_DISABLED=Off
Telescope Simulator.DEVICE_PORT_SCAN.Scan Ports=Off
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_GPS=GPS Simulator
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_DOME=Dome Simulator
Telescope Simulator.DOME_POLICY.DOME_IGNORED=On
Telescope Simulator.DOME_POLICY.DOME_LOCKS=Off
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_FOCAL_LENGTH=900
Telescope Simulator.TELESCOPE_INFO.GUIDER_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.GUIDER_FOCAL_LENGTH=900
Telescope Simulator.SCOPE_CONFIG_NAME.SCOPE_CONFIG_NAME=
Telescope Simulator.MOUNT_TYPE is write-only
Telescope Simulator.SIM_PIER_SIDE is write-only
Telescope Simulator.MOUNT_MODEL is write-only
Telescope Simulator.MOUNT_AXES.PRIMARY=0
Telescope Simulator.MOUNT_AXES.SECONDARY=0
Telescope Simulator.FLIP_HA is write-only
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_GPS=GPS Simulator
Telescope Simulator.ACTIVE_DEVICES.ACTIVE_DOME=Dome Simulator
Telescope Simulator.DOME_POLICY.DOME_IGNORED=On
Telescope Simulator.DOME_POLICY.DOME_LOCKS=Off
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.TELESCOPE_FOCAL_LENGTH=900
Telescope Simulator.TELESCOPE_INFO.GUIDER_APERTURE=120
Telescope Simulator.TELESCOPE_INFO.GUIDER_FOCAL_LENGTH=900
Telescope Simulator.SCOPE_CONFIG_NAME.SCOPE_CONFIG_NAME=
Telescope Simulator.MOUNT_TYPE is write-only
Telescope Simulator.SIM_PIER_SIDE is write-only
Telescope Simulator.MOUNT_MODEL is write-only
Telescope Simulator.MOUNT_AXES.PRIMARY=0
Telescope Simulator.MOUNT_AXES.SECONDARY=0
Telescope Simulator.FLIP_HA is write-only

libcamera: not detected

libcamera packages
ii  libcamera-apps-lite                  1.2.1-1                          arm64        libcamera-apps-lite
ii  libcamera0:arm64                     0~git20230720+bde9b04f-1         arm64        complex camera support library
ii  python3-libcamera                    0~git20230720+bde9b04f-1         arm64        complex camera support library (Python 3 bindings)

libcamera cameras
libcamera-hello: /usr/bin/libcamera-hello
No cameras available!

python packages
ii  libpython3-dev:arm64                 3.9.2-3                          arm64        header files and a static library for Python (default)
ii  libpython3-stdlib:arm64              3.9.2-3                          arm64        interactive high-level object-oriented language (default python3 version)
ii  libpython3.9:arm64                   3.9.2-1                          arm64        Shared Python runtime library (version 3.9)
ii  libpython3.9-dev:arm64               3.9.2-1                          arm64        Header files and a static library for Python (v3.9)
ii  libpython3.9-minimal:arm64           3.9.2-1                          arm64        Minimal subset of the Python language (version 3.9)
ii  libpython3.9-stdlib:arm64            3.9.2-1                          arm64        Interactive high-level object-oriented language (standard library, version 3.9)
ii  python-apt-common                    2.2.1                            all          Python interface to libapt-pkg (locales)
ii  python-is-python3                    3.9.2-1                          all          symlinks /usr/bin/python to python3
ii  python-pip-whl                       20.3.4-4+rpt1+deb11u1            all          Python package installer (pip wheels)
ii  python3                              3.9.2-3                          arm64        interactive high-level object-oriented language (default python3 version)
ii  python3-appdirs                      1.4.4-1                          all          determining appropriate platform-specific directories (Python 3)
ii  python3-apt                          2.2.1                            arm64        Python 3 interface to libapt-pkg
ii  python3-certifi                      2020.6.20-1                      all          root certificates for validating SSL certs and verifying TLS hosts (python3)
ii  python3-chardet                      4.0.0-1                          all          universal character encoding detector for Python3
ii  python3-colorzero                    1.1                              all          Construct, convert, and manipulate colors in a Pythonic manner.
ii  python3-debconf                      1.5.77                           all          interact with debconf from Python 3
ii  python3-dev                          3.9.2-3                          arm64        header files and a static library for Python (default)
ii  python3-distlib                      0.3.2+really+0.3.1-0.1           all          low-level components of python distutils2/packaging
ii  python3-distro                       1.5.0-1                          all          Linux OS platform information API
ii  python3-distutils                    3.9.2-1                          all          distutils package for Python 3.x
ii  python3-filelock                     3.0.12-2                         all          platform independent file locking module
ii  python3-gpiozero                     1.6.2-1                          all          Simple API for controlling devices attached to a Pi's GPIO pins.
ii  python3-idna                         2.10-1                           all          Python IDNA2008 (RFC 5891) handling (Python 3)
ii  python3-importlib-metadata           1.6.0-2                          all          library to access the metadata for a Python package - Python 3.x
ii  python3-kms++                        0~git20221104~171243+adc05b6-1   arm64        C++ library for kernel mode setting - python3 bindings
ii  python3-lib2to3                      3.9.2-1                          all          Interactive high-level object-oriented language (lib2to3)
ii  python3-libcamera                    0~git20230720+bde9b04f-1         arm64        complex camera support library (Python 3 bindings)
ii  python3-minimal                      3.9.2-3                          arm64        minimal subset of the Python language (default python3 version)
ii  python3-more-itertools               4.2.0-3                          all          library with routines for operating on iterables, beyond itertools (Python 3)
ii  python3-numpy                        1:1.19.5-1                       arm64        Fast array facility to the Python 3 language
ii  python3-picamera2                    0.3.12-2                         all          Libcamera based python library
ii  python3-pidng                        4.0.9-1                          arm64        Create Adobe DNG RAW files using Python (Python 3)
ii  python3-piexif                       1.1.3-1                          all          Exif manipulation with pure python script (Python 3)
ii  python3-pil:arm64                    8.1.2+dfsg-0.3+deb11u1           arm64        Python Imaging Library (Python3)
ii  python3-pip                          20.3.4-4+rpt1+deb11u1            all          Python package installer
ii  python3-pkg-resources                52.0.0-4                         all          Package Discovery and Resource Access using pkg_resources
ii  python3-prctl                        1.7-2+b4                         arm64        Python interface to the prctl() syscall (Python 3)
ii  python3-requests                     2.25.1+dfsg-2                    all          elegant and simple HTTP library for Python3, built for human beings
ii  python3-rpi.gpio                     0.7.0-0.2+b2                     arm64        Module to control Raspberry Pi GPIO channels (Python 3)
ii  python3-setuptools                   52.0.0-4                         all          Python3 Distutils Enhancements
ii  python3-simplejpeg                   1.6.4-1                          arm64        Simple package for fast JPEG encoding and decoding
ii  python3-six                          1.16.0-2                         all          Python 2 and 3 compatibility library (Python 3 interface)
ii  python3-spidev                       20200602~200721-1                arm64        Bindings for Linux SPI access through spidev (Python 3)
ii  python3-tk:arm64                     3.9.2-1                          arm64        Tkinter - Writing Tk applications with Python 3.x
ii  python3-toml                         0.10.1-1                         all          library for Tom's Obvious, Minimal Language - Python 3.x
ii  python3-urllib3                      1.26.5-1~exp1                    all          HTTP library with thread-safe connection pooling for Python3
ii  python3-v4l2                         0.3.2-1                          all          Python bindings for the v4l2 userspace api
ii  python3-venv                         3.9.2-3                          arm64        venv module for python3 (default python3 version)
ii  python3-virtualenv                   20.4.0+ds-2+deb11u1              all          Python virtual environment creator
ii  python3-wheel                        0.34.2-1                         all          built-package format for Python
ii  python3-zipp                         1.0.0-3                          all          pathlib-compatible Zipfile object wrapper - Python 3.x
ii  python3.9                            3.9.2-1                          arm64        Interactive high-level object-oriented language (version 3.9)
ii  python3.9-dev                        3.9.2-1                          arm64        Header files and a static library for Python (v3.9)
ii  python3.9-minimal                    3.9.2-1                          arm64        Minimal subset of the Python language (version 3.9)
ii  python3.9-venv                       3.9.2-1                          arm64        Interactive high-level object-oriented language (pyvenv binary, version 3.9)

Detected indi-allsky virtualenv
virtualenv python: Python 3.9.2
virtualenv PATH: /home/sky/indi-allsky/virtualenv/indi-allsky/bin:/usr/local/bin:/usr/bin:/bin
flask command: /home/sky/indi-allsky/virtualenv/indi-allsky/bin/flask

virtualenv python modules
alembic==1.13.1
apache-libcloud==3.8.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
asciitree==0.3.3
astroalign==2.5.1
astropy==6.0.0
astropy-healpix==1.0.2
astropy-iers-data==0.2024.1.22.0.30.30
astroscrappy==1.1.0
bcrypt==4.1.2
blinker==1.7.0
boto3==1.34.29
botocore==1.34.29
Bottleneck==1.3.7
cachetools==5.3.2
ccdproc==2.4.1
certifi==2023.11.17
cffi==1.16.0
charset-normalizer==3.3.2
circuitbreaker==1.4.0
click==8.1.7
cloudpickle==3.0.0
cryptography==41.0.7
Cython==3.0.8
dask==2024.1.1
dbus-python==1.3.2
ephem==4.1.5
fasteners==0.19
Flask==3.0.1
Flask-Login==0.6.3
Flask-Migrate==4.0.5
Flask-SQLAlchemy==3.1.1
Flask-WTF==1.2.1
fsspec==2023.12.2
google-api-core==2.15.0
google-api-python-client==2.115.0
google-auth==2.27.0
google-auth-httplib2==0.2.0
google-auth-oauthlib==1.2.0
google-cloud-core==2.4.1
google-cloud-storage==2.14.0
google-crc32c==1.5.0
google-resumable-media==2.7.0
googleapis-common-protos==1.62.0
greenlet==3.0.3
gunicorn==21.2.0
httplib2==0.22.0
idna==3.6
imageio==2.33.1
imageio-ffmpeg==0.4.9
importlib-metadata==7.0.1
inotify==0.2.10
is-safe-url==1.0
itsdangerous==2.1.2
Jinja2==3.1.3
jmespath==1.0.1
lazy_loader==0.3
locket==1.0.0
lxml==5.1.0
Mako==1.3.0
MarkupSafe==2.1.4
mysql-connector-python==8.3.0
networkx==3.2.1
nose==1.3.7
numcodecs==0.12.1
numpy==1.26.3
oauthlib==3.2.2
oci==2.119.1
opencv-python-headless==4.9.0.80
packaging==23.2
paho-mqtt==1.6.1
paramiko==3.4.0
partd==1.4.1
passlib==1.7.4
piexif==1.1.3
pillow==10.2.0
pkg_resources==0.0.0
prettytable==3.9.0
protobuf==4.25.2
psutil==5.9.8
pyasn1==0.5.1
pyasn1-modules==0.3.0
pycparser==2.21
pycurl==7.45.2
pyerfa==2.0.1.1
pygifsicle==1.0.7
pyindi-client @ git+https://github.com/indilib/pyindi-client.git@674706fc4caca8c28ab60ada121b8554027a56c7
PyMySQL==1.1.0
PyNaCl==1.5.0
pyOpenSSL==23.3.0
pyparsing==3.1.1
python-dateutil==2.8.2
pytz==2023.3.post1
PyYAML==6.0.1
rawpy==0.19.0
reproject==0.13.0
requests==2.31.0
requests-oauthlib==1.3.1
requests-toolbelt==1.0.0
rsa==4.9
s3transfer==0.10.0
scikit-image==0.22.0
scipy==1.12.0
semantic-version==2.10.0
sep==1.2.1
setuptools-rust==1.8.1
shapely==2.0.2
six==1.16.0
SQLAlchemy==2.0.25
support-developer==1.0.5
tifffile==2023.12.9
tomli==2.0.1
toolz==0.12.1
typing_extensions==4.9.0
uritemplate==4.1.1
urllib3==1.26.18
wcwidth==0.2.13
Werkzeug==3.0.1
WTForms==3.1.2
zarr==2.16.1
zipp==3.17.0

#################################
###     end support info      ###
#################################
sky@Titis:~/indi-allsky $

**Comment by aaronwmorris** [comment]
This is the problem I figured it was.  The issue is the fxload utility that comes with Debian 10/11 that is used to load the firmware on the QHY cameras does not support USB3 targets.  The fix is building a new version that does support USB3.

This page will walk you through the steps:
https://github.com/aaronwmorris/indi-allsky/wiki/Build-fxload-for-USB3-support

This is the device that exists to load the firmware for the QHY camera:
```
Bus 001 Device 003: ID 1618:0462 Cypress WestBridge
```

The device you need to target in the command for the QHY 5III 462 is `-d 1618:0462`

**Comment by Astrokashyap** [comment]
I did try that but, cant manage to run `  sudo install -o root -g root -m 755 ./.libs/fxload /usr/local/sbin`  
let me know what stupid thing did i miss
![qhy3](https://github.com/aaronwmorris/indi-allsky/assets/157868841/61d2e743-a629-4e69-b76e-9e9daeaa6a2a)

**Reply by aaronwmorris** [reply]
It looks like everything there was successful.  You should be able to try to manually load the firmware now.

This is the command pointing to the 462 firmware and the correct usb device ID:
```
sudo LD_LIBRARY_PATH=/usr/local/lib /usr/local/sbin/fxload -t fx3 -I /lib/firmware/qhy/QHY5III462.img -d 1618:0462
```

If this works, you should be able to see the QHY camera with `lsusb`

**Reply by Astrokashyap** [reply]
Yes thanks it worked, but on rebooting again the same thing happened with`lsusb`  saw the same 

Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 1618:0462 Cypress WestBridge
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub.

running `sudo LD_LIBRARY_PATH=/usr/local/lib /usr/local/sbin/fxload -t fx3 -I /lib/firmware/qhy/QHY5III462.img -d 1618:0462` again gave :--
![qhy4](https://github.com/aaronwmorris/indi-allsky/assets/157868841/29ea930b-229c-4748-b8ad-14052a97a22f)

**Reply by aaronwmorris** [reply]
The last section of the wiki deals with automatically loading the firmware for the camera.

https://github.com/aaronwmorris/indi-allsky/wiki/Build-fxload-for-USB3-support

You just need to make sure you use the right `idVendor`, `idProduct`, and firmware in the udev config.
